packages/itwin/measure-tools/src/ui-2.0/MeasureToolsUiProvider.tsx

```diff
@@ -48,7 +53,10 @@ export class MeasureToolsUiItemsProvider implements UiItemsProvider {
       },
       enableSheetMeasurement: props?.enableSheetMeasurement ?? false,
       stageUsageList: props?.stageUsageList ?? [StageUsage.General],
+      onFeatureUsed: props?.onFeatureUsed,
     };
+    if (!FeatureTracking.onFeature.numberOfListeners && this._props.onFeatureUsed)
```

Comment:
I'd nix the condition checking the number of listenes.

@aruniverse Do UI providers ever get cleaned up? Or only ever created once per app lifetime?

Reply:
The reason for `!FeatureTracking.onFeature.numberOfListeners` is as I test on a sample viewer this listener is added twice. I guess the UI provider might get cleaned up and we have not removed this listener before re-creating the UI provider (but I am not really clear on when it gets cleaned up or why it might be created twice).

Reply:
With FeatureTracking.onFeature being public, I don't think it's appropriate to assume that nobody else is listening for that event. I doesn't seem acceptable to have this not work because the event is already being listened to by somebody else.

Reply:
Looks like the interface has a cleanup mechanism: https://www.itwinjs.org/reference/appui-react/uiprovider/uiitemsprovider/onunregister/

TBH something like this belongs better in the `MeasureTools.startup` which the app has to call anyways. It also has an associated terminate, which should be called alongside the IModelApp startup/shutdown by the app.

Reply:
Sure, I can implement it as a startup option instead. However, its usage will be different from [tree-widget](https://www.npmjs.com/package/@itwin/tree-widget-react) package (where you can provide onFeatureUsed callback function to UI Provider). If you are fine with that, I will do it right away. Thank you!